modules/gapi/test/gapi_sample_pipelines_srlz.cpp

```diff
@@ -0,0 +1,298 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#include "test_precomp.hpp"
+
+#include <stdexcept>
+#include <ade/util/iota_range.hpp>
+#include "logger.hpp"
+
+#include <opencv2/gapi/core.hpp>
+#include "serialization/core.hpp"
+#include "serialization/imgproc.hpp"
+
+namespace opencv_test
+{
+TEST(GAPI_Pipeline, CropSerialization_Rect)
+{
+    cv::Rect rect_to{ 4,10,37,50 };
+    cv::Size sz_in = cv::Size(1920, 1080);
+    cv::Size sz_out = cv::Size(37, 50);
+    cv::Mat out_mat_gapi(sz_out, CV_8UC1);
+    cv::Mat out_mat_ocv(sz_out, CV_8UC1);
+    cv::Mat in_mat = cv::Mat::eye(sz_in, CV_8UC1);
+
+
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::GMat in;
+    auto out = cv::gapi::crop(in, rect_to);
+    cv::gapi::GKernelPackage srlz_kernels = combine(cv::gapi::core::serialization::kernels(), cv::gapi::imgproc::serialization::kernels());
+    cv::GComputation c(in, out);
+    c.apply(in_mat, out_mat_gapi, cv::GCompileArgs(cv::compile_args(srlz_kernels)));
+
+    // OpenCV code /////////////////////////////////////////////////////////////
+    {
+        cv::Mat(in_mat, rect_to).copyTo(out_mat_ocv);
```

Comment:
btw there's no even need in a `copyTo`, just `out_mat_ocv = in_mat(rect_to)`.

Reply:
Done